### PR TITLE
Update the documentation and examples to match the current state

### LINKED ISSUE #102: Document the basic usage of Nickel-nix
Update the README to show how to use Nickel-nix (at least in the happy path)

### PR DESCRIPTION
**Pull request by thufschmitt**
- Change all the self references to `organist` (except for the cachix bucket which is more complex)
- Update the README based on the “ideal” one to match the current state

Fix #102

### CODE REVIEW

**Review by YorikSar — CHANGES_REQUESTED** [review]

File: doc/dependency-management.md [hunk 1]
```diff
@@ -1,53 +1,49 @@
...
   },
+
   shells.dev = {
-    packages = [],
+    packages.hello = organist.lib.import_nix "nixpkgs#hello",
   },
-
-  # More stuff
-}
+} | organist.contracts.NixelExpression
```
[the remark this answers is not in the document]
Comment (YorikSar) [reply doc/dependency-management.md:23]: Actually, with this https://github.com/nickel-lang/organist/blob/main/lib/lib.nix#L89-L90 we don't need this at all, do we?

[the remark this answers is not in the document]
Comment (YorikSar) [reply doc/dependency-management.md:23]: Link refers to this part of `lib.nix`:

```
      let nickel_expr | nix.contracts.NixelExpression =
        import "${sources}/${nickelFile}" in
```

[the remark this answers is not in the document]
Comment (thufschmitt) [reply doc/dependency-management.md]: It's not strictly required for the evaluation, but it gives the LSP some context, which is very useful.

But yeah, the name should be changed, I'll do that right away

File: run-test.sh [hunk 2]
```diff
@@ -42,7 +42,7 @@ test_one_template () (
   pushd ./example
   sed -i "s/shells\.Bash/shells.$target/" project.ncl
   prepare_shell
-  nix develop --accept-flake-config --print-build-logs < /dev/null
+  nix develop --accept-flake-config --print-build-logs
```
[the remark this answers is not in the document]
Comment (thufschmitt) [reply run-test.sh]: Indeed that was committed by accident, thanks. Fixed

**Review by YorikSar — APPROVED** [review]

**Pull request merged**